pkg/crd/cloudprivateipconfig/v1/types.go

```diff
@@ -0,0 +1,62 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +genclient:noStatus
+// +resource:path=cloudprivateipconfig
+// +kubebuilder:resource:shortName=cpip,scope=Cluster
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:printcolumn:name="Node Request",type=string,JSONPath=".spec.items[*].node"
+// +kubebuilder:printcolumn:name="IP Request",type=string,JSONPath=".spec.items[*].ip"
+// +kubebuilder:printcolumn:name="Node Assignment",type=string,JSONPath=".status.items[*].node"
+// +kubebuilder:printcolumn:name="IP Assignment",type=string,JSONPath=".status.items[*].ip"
+// CloudPrivateIPConfig is a CRD allowing the user to assign private
+// IP addresses to the primary NIC on cloud VMs. This is done by
```

Comment:
nit: verbiage refinement.
Suggested verbiage: 
`CloudPrivateIPConfig is a CRD type that allows users to create a config request for assigning private IP addresses to a NIC associated with a cloud VM.` 
The part about how to do it can be in the type of the `Spec` field

Reply:
The CR is not user facing right ? would only allow ovnkube-master to create the CR right ?

Reply:
> The CR is not user facing right ?

Yes

> would only allow ovnkube-master to create the CR right ?

We could make it so that only the service accounts associated with the network plugins are allowed to create the CR.  

I however agree with @abhat that this sentence could benefit from some re-writing.